### PR TITLE
feat(vscode): Add New Code Project Command 

### PR DESCRIPTION
**Pull request by samikay101**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -56,4 +56,5 @@ __azurite_db_queue_extent__.json
 
 # Generated Docusaurus files
 .docusaurus/
-.cache-loader/
\ No newline at end of file
+.cache-loader/
+.fake
```
Comment (ccastrotrejo) [inline .gitignore:60]: Is this change needed?

File: apps/vs-code-designer/src/app/commands/workflows/openOverview.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ export async function openOverview(context: IAzureConnectorsContext, node: vscod
     panelName = `${vscode.workspace.name}-${workflowName}-overview`;
     workflowContent = JSON.parse(readFileSync(workflowFilePath, 'utf8'));
     baseUrl = `http://localhost:${ext.workflowRuntimePort}${managementApiPrefix}`;
-    apiVersion = '2019-10-01-edge-preview';
+    apiVersion = '2018-11-01';
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/workflows/openOverview.ts:54]: If this is working fine on Swechha environment, lets not update this until we are really sure this will fix it

File: apps/vs-code-designer/src/app/commands/initProjectForVSCode/InitVSCodeStepBase.ts [hunk 3]
```diff
@@ -81,7 +80,7 @@ export abstract class InitVSCodeStepBase extends AzureWizardExecuteStep<IProject
 
     context.telemetry.properties.isProjectInSubDir = String(isSubpath(context.workspacePath, context.projectPath));
 
-    const vscodePath: string = path.join(context.workspacePath, vscodeFolderName);
+    const vscodePath: string = path.join(context.workspacePath, '.vscode');
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/initProjectForVSCode/InitVSCodeStepBase.ts:83]: I feel like you still need to update your branch with the latest changes of main. This was a change  I made 10 days ago and it seems like it is getting reverted here.

File: apps/vs-code-designer/src/app/commands/createNewProject/createProjectSteps/FolderListStep.ts [hunk 4]
```diff
@@ -1,7 +1,3 @@
-/*---------------------------------------------------------------------------------------------
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewProject/createProjectSteps/FolderListStep.ts:1]: There is no need to remove this

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/OpenFolderStepCodeProject.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import { OpenBehavior } from '@microsoft/vscode-extension';
+import * as fs from 'fs';
+import * as path from 'path';
+import { commands, Uri, workspace } from 'vscode';
+
+/**
+ * A class that extends AzureWizardExecuteStep and is responsible for opening a folder in VS Code.
+ */
+export class OpenFolderStepCodeProject extends AzureWizardExecuteStep<IProjectWizardContext> {
+  public priority = 250;
```
Comment (mireedmsft) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/OpenFolderStepCodeProject.ts:16]: nit: consider adding a comment here to explain where you get this number.

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/InvokeFunctionProjectSetup.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    return true;
+  }
+
+  /**
+   * Creates the .cs file inside the functions folder.
+   * @param functionFolderPath The path to the functions folder.
+   * @param methodName The name of the method.
+   * @param namespace The name of the namespace.
+   */
+  private async createCsFile(functionFolderPath: string, methodName: string, namespace: string): Promise<void> {
```
Comment (mireedmsft) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/InvokeFunctionProjectSetup.ts:52]: is it a normal pattern to check a template in to code like this? In a .net project this is usually done with a T4 template that is then generated at build time, but not I'm not sure what the standard pattern is for TypeScript

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/OpenBehaviorStepCodeProject.ts
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/OpenBehaviorStepCodeProject.ts:1]: Is there any difference in this and the apps/vs-code-designer/src/app/commands/createNewProject/OpenBehaviorStep.ts file? if not, maybe just use the one that is already there,

**Review by ccastrotrejo — CHANGES_REQUESTED** [review]

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createNewCodeProject.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  context: IActionContext,
+  folderPath?: string | undefined,
+  language?: ProjectLanguage,
+  version?: ProjectVersion,
+  openFolder = true,
+  templateId?: string,
+  functionName?: string,
+  functionSettings?: { [key: string]: string | undefined }
+): Promise<void> {
+  await createNewCodeProjectInternal(context, {
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createNewCodeProject.ts:35]: NIT: I feel that you don't need to follow this from the other NewCodeProject command, by directly having the call to the other function or having all the logic from the other function in this should work fine

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/ScriptProjectCreateStep.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    // Create the local.settings.json file
+    const localSettingsJsonPath: string = path.join(context.projectPath, localSettingsFileName);
+    const logicAppFolderPath = context.logicAppFolderPath;
+    if (await confirmOverwriteFile(context, localSettingsJsonPath)) {
+      const localSettingsJson: ILocalSettingsJson = {
+        IsEncrypted: false,
+        Values: {
+          AzureWebJobsStorage: 'UseDevelopmentStorage=true',
+          FUNCTIONS_WORKER_RUNTIME: 'dotnet-isolated',
+          WORKFLOWS_SUBSCRIPTION_ID: '',
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/ScriptProjectCreateStep.ts:64]: Lets not add this value, lets let the user make the decision whether he wants to use azure connectors or no

File: apps/vs-code-designer/src/app/commands/workflows/openDesigner/openDesignerForLocalProject.ts [hunk 9]
```diff
@@ -42,7 +42,7 @@ export default class OpenDesignerForLocalProject extends OpenDesignerBase {
 
   constructor(context: IActionContext, node: Uri) {
     const workflowName = path.basename(path.dirname(node.fsPath));
-    const apiVersion = '2018-11-01';
+    const apiVersion = '2019-10-01-edge-preview';
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/workflows/openDesigner/openDesignerForLocalProject.ts:45]: Can you please revert this as well?

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/InvokeFunctionProjectSetup.ts (hunk 6 above)
Comment (ccastrotrejo) [reply apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/InvokeFunctionProjectSetup.ts:52]: Yeah, lets check if there is a way to store the template and bring it in. Not having the whole txt over there

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/ScriptProjectCreateStep.ts [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+    // Create the local.settings.json file
+    const localSettingsJsonPath: string = path.join(context.projectPath, localSettingsFileName);
+    const logicAppFolderPath = context.logicAppFolderPath;
+    if (await confirmOverwriteFile(context, localSettingsJsonPath)) {
+      const localSettingsJson: ILocalSettingsJson = {
+        IsEncrypted: false,
+        Values: {
+          AzureWebJobsStorage: 'UseDevelopmentStorage=true',
+          FUNCTIONS_WORKER_RUNTIME: 'dotnet-isolated',
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/ScriptProjectCreateStep.ts:63]: Let's follow that src/commands/createNewProject/ProjectCreateStep/ScriptProjectCreateStep.ts line 44 follows on the functions worker runtime

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/CodelessFunctiontWorkflow.ts [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+
+  // Async method that creates a new workflow for the codeless Azure Function project
+  public async executeCore(context: IFunctionWizardContext): Promise<string> {
+    // Get the function template and function name from the IFunctionWizardContext object
+    const template: IWorkflowTemplate = nonNullProp(context, 'functionTemplate');
+    const functionPath: string = path.join(context.projectPath, nonNullProp(context, 'functionName'));
+    const methodName = context.methodName;
+
+    // Create empty stateful and stateless definition objects
+    const emptyStatefulDefinition: StandardApp = {
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/CodelessFunctiontWorkflow.ts:59]: I can't remember, can you make sure if it is desire to have an already created schema or an empty schema its fine

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/WorkflowCode.ts [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+        args: ['${input:getDebugSymbolDll}'],
+        type: 'process',
+        problemMatcher: '$msCompile',
+      },
+      {
+        type: func,
+        command: hostStartCommand,
+        problemMatcher: funcWatchProblemMatcher,
+        isBackground: true,
+        label: 'func: host start',
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/WorkflowCode.ts:29]: NIT: we have a `hostStartTaskName` constant for this.

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/OpenFolderStepCodeProject.ts [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+    // Check if user has selected to add folder to workspace and update workspace accordingly
+    if (context.openBehavior === OpenBehavior.addToWorkspace && openFolders.length === 0) {
+      context.openBehavior = OpenBehavior.openInCurrentWindow;
+    }
+
+    if (context.openBehavior === OpenBehavior.addToWorkspace) {
+      workspace.updateWorkspaceFolders(openFolders.length, 0, { uri: uri });
+    } else {
+      // Open folder using executeCommand method of commands object with vscode.openFolder command
+      await commands.executeCommand('vscode.openFolder', uri, context.openBehavior === OpenBehavior.openInNewWindow /* forceNewWindow */);
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/OpenFolderStepCodeProject.ts:45]: 👏🏽

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/FunctionConfigFile.ts [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+  }
+
+  /**
+   * Generates the extensions.json file in the specified folder.
+   * @param folderPath The path to the folder where the extensions.json file should be generated.
+   */
+  private async generateExtensionsJson(folderPath: string): Promise<void> {
+    const filePath = path.join(folderPath, 'extensions.json');
+    const content = {
+      recommendations: ['ms-azuretools.vscode-azurefunctions', 'ms-dotnettools.csharp'],
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/FunctionConfigFile.ts:54]: NIT: I don't think this is needed as a recommendation since we move the azure functions as a dependency, so it will be automatically downloaded when our extension gets downloaded.  For the csharp extension maybe add that string into the constants file

**Review by ccastrotrejo — APPROVED** [review]

**Comment by samikay101** [comment]
#assign @ecfan

**Comment by ecfan** [comment]
@samikay101: Hi, what file am I supposed to review?

File: apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsProj [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+    <OutputType>Library</OutputType>
+    <PlatformTarget>x64</PlatformTarget>
+    <!-- Please replace 'LogicAppFolder' with the name of your folder that contains your logic app project. -->
+    <LogicAppFolder>LogicApp</LogicAppFolder>
+    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+ </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Please ensure you have the 'Microsoft.Azure.Functions.Extensions.Workflows.WorkflowActionTrigger' package and the 'Microsoft.Azure.WebJobs.Core' dependency. -->
+    <PackageReference Include="Microsoft.Azure.Functions.Extensions.Workflows.WorkflowActionTrigger" Version="1.0.0" />
```
Comment (rohithah) [inline apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsProj:15]: doesnt this need to be "Microsoft.Azure.Workflows.Webjobs.Sdk"?

Comment (rohithah) [reply apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsProj:15]: and the version is 1.0.0-preview"

File: apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsFile [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+                }
+            };
+
+            return Task.FromResult(result); 
+        }
+
+        /// <summary>
+        /// The wrapper.
+        /// </summary>
+        public class Wrapper
```
Comment (rohithah) [inline apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsFile:41]: this file seems to be outdated

Comment (rohithah) [reply apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsFile:41]: looks like its commented out - can we actually remove the commented code?

File: apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsFile [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+                DayLow = String.Format("The low for the day is {0} {1}", lowTemp, temperatureScale),
+                DayHigh = String.Format("The high for the day is {0} {1}", highTemp, temperatureScale)
+            };
+
+            return Task.FromResult(weather);
+        }
+
+        public class Weather
+        {
+            /// <summary>
```
Comment (rohithah) [inline apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsFile:72]: I think we should remove this summary comment (talks about worker, an implementation detail)

File: apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsProj (hunk 15 above)
Comment (rohithah) [reply apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsProj:15]: i still see the older version - am I missing something?

File: apps/vs-code-designer/src/package.json [hunk 18]
```diff
@@ -67,6 +67,15 @@
           "dark": "assets/dark/CreateNewProject.svg"
         }
       },
+      {
+        "command": "azureLogicAppsStandard.createNewCodeProject",
+        "title": "Create New Code Project...",
```
Comment (rohithah) [inline apps/vs-code-designer/src/package.json:72]: "Create Logic App workspace"?

Comment (ecfan) [reply apps/vs-code-designer/src/package.json:72]: Why change "project" to "workspace"? Eventually, you deploy the project, not a workspace? Also, why do we have `azureLogicAppsStandard.createNewProject` and `azureLogicAppsStandard.createNewCodeProject`? What's the difference?

Let's minimize the sentence casing "Logic App" because we don't want people to think that "logic app" is capitalized. Either "logic app" (the resource) or "Azure Logic Apps" (the service).

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/InitCodeProject.ts [hunk 19]
```diff
@@ -0,0 +1,370 @@
...
+    folder: WorkspaceFolder | undefined,
+    vscodePath: string,
+    version: FuncVersion
+  ): Promise<void> {
+    if (this.getDebugConfiguration) {
+      const newDebugConfig: DebugConfiguration = this.getDebugConfiguration(version);
+      const versionMismatchError: Error = new Error(
+        localize(
+          'versionMismatchError',
+          'The version in your {0} must be "{1}" to work with Azure Functions.',
```
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/InitCodeProject.ts:270]: What values are shown by the placeholders?

File: apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsFile [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+            /// </summary>
+            public Dictionary<string, object> RandomProperty { get; set; }
+        }
+        */
+
+        [FunctionName("<%= methodName %>")]
+        public static Task<Weather> Run([WorkflowActionTrigger] int zipCode, string temperatureScale)
+        {
+            Random rnd = new Random();
+            var currentTemp = temperatureScale == "Celcius" ? rnd.Next(1, 30) : rnd.Next(40, 90);
```
Comment (ecfan) [inline apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsFile:54]:
```suggestion
            var currentTemp = temperatureScale == "Celsius" ? rnd.Next(1, 30) : rnd.Next(40, 90);
```

File: apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsProj [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
+    <OutputType>Library</OutputType>
+    <PlatformTarget>x64</PlatformTarget>
+    <!-- Please replace 'LogicAppFolder' with the name of your folder that contains your logic app project. -->
+    <LogicAppFolder>LogicApp</LogicAppFolder>
+    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+ </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Please ensure you have the 'Microsoft.Azure.Functions.Extensions.Workflows.WorkflowActionTrigger' package and the 'Microsoft.Azure.WebJobs.Core' dependency. -->
```
Comment (ecfan) [inline apps/vs-code-designer/src/assets/FunctionProjectTemplate/FunctionsProj:14]:
```suggestion
    <!-- Make sure that you have the 'Microsoft.Azure.Functions.Extensions.Workflows.WorkflowActionTrigger' package and the 'Microsoft.Azure.WebJobs.Core' dependency. -->
```

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/SetWorkspaceName.ts [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+import type { IProjectWizardContext } from '@microsoft/vscode-extension';
+import * as path from 'path';
+
+export class setWorkspaceName extends AzureWizardPromptStep<IProjectWizardContext> {
+  public hideStepCount = true;
+
+  public async prompt(context: IProjectWizardContext): Promise<void> {
+    context.workspaceName = await context.ui.showInputBox({
+      placeHolder: localize('setWorkspaceName', 'Workspace name'),
+      prompt: localize('workspaceNamePrompt', 'Provide a Workspace name'),
```
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/SetWorkspaceName.ts:17]:
```suggestion
      prompt: localize('workspaceNamePrompt', 'Provide a workspace name'),
```

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/setMethodName.ts [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import type { IProjectWizardContext } from '@microsoft/vscode-extension';
+
+export class setMethodName extends AzureWizardPromptStep<IProjectWizardContext> {
+  public hideStepCount = true;
+
+  public async prompt(context: IProjectWizardContext): Promise<void> {
+    context.methodName = await context.ui.showInputBox({
+      placeHolder: localize('setMethodName', 'Method name'),
+      prompt: localize('methodNamePrompt', 'Provide a Function Name for Logic App Project'),
```
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/setMethodName.ts:15]:
```suggestion
      prompt: localize('methodNamePrompt', 'Provide a function name for logic app project'),
```

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/setNamepSpace.ts [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import type { IProjectWizardContext } from '@microsoft/vscode-extension';
+
+export class setNameSpace extends AzureWizardPromptStep<IProjectWizardContext> {
+  public hideStepCount = true;
+
+  public async prompt(context: IProjectWizardContext): Promise<void> {
+    context.namespaceName = await context.ui.showInputBox({
+      placeHolder: localize('setNameSpace', 'nameSpace'),
+      prompt: localize('methodNamePrompt', 'Provide a Namespace for Logic App Project'),
```
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/setNamepSpace.ts:15]:
```suggestion
      prompt: localize('methodNamePrompt', 'Provide a namespace for logic app project'),
```

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/InitCodeProject.ts [hunk 25]
```diff
@@ -0,0 +1,370 @@
...
+  protected settings: ISettingToAdd[] = [];
+
+  protected abstract executeCore(context: IProjectWizardContext): Promise<void>;
+  protected abstract getTasks(): TaskDefinition[];
+  protected getTaskInputs?(): ITaskInputs[];
+  protected getWorkspaceSettings?(): ISettingToAdd[];
+
+  protected getDebugConfiguration(version: FuncVersion): DebugConfiguration {
+    return {
+      name: localize('attachToNetFunc', 'Attach to Logic App'),
```
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/InitCodeProject.ts:69]: Per [Microsoft Style Guide - Capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization): 

```suggestion
      name: localize('attachToNetFunc', 'Attach to logic app'),
```

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/InitCodeProject.ts [hunk 26]
```diff
@@ -0,0 +1,370 @@
...
+  protected addSubDir(context: IProjectWizardContext, fsPath: string): string {
+    const subDir: string = path.relative(context.workspacePath, context.projectPath);
+    // always use posix for debug config
+    return path.posix.join(subDir, fsPath);
+  }
+
+  private async writeTasksJson(context: IProjectWizardContext, vscodePath: string): Promise<void> {
+    const newTasks: TaskDefinition[] = this.getTasks();
+    const versionMismatchError: Error = new Error(
+      localize('versionMismatchError', 'The version in your {0} must be "{1}" to work with Azure Functions.', tasksFileName, tasksVersion)
```
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/initLogicAppCodeProjectVScode/InitCodeProject.ts:173]: What values do the placeholders represent?

**Comment by samikay101** [comment]
Resolves AB#19266628
Resolves AB#2427
